ghpages.mk

```
@@ -77,6 +77,15 @@ cleanup-ghpages: $(GHPAGES_ROOT)
 	  git remote prune $$remote; \
 	done;
 
+# Drop old gh-pages commits (keep 30-60 days of history)
+	@CUTOFF_30=$$(($$(date '+%s')-2592000)); \
+	CUTOFF_60=$$(($$(date '+%s')-5184000)); \
+	ROOT=$$(git -C $(GHPAGES_TARGET) rev-list --max-parents=0 gh-pages); \
+	if [ `git -C $(GHPAGES_ROOT) show -s --format=%ct $$ROOT` -lt $$CUTOFF_60 ]; then \
+	  git -C $(GHPAGES_ROOT) replace --graft `git rev-list --min-age=$$CUTOFF_30 --max-count=1 gh-pages`; \
+	  git -C $(GHPAGES_ROOT) filter-branch gh-pages; \
```

```suggestion
	@KEEP=$$((`date '+%s'`-2592000)); CUTOFF=$$((`date '+%s'`-5184000)); \
	ROOT=`git -C $(GHPAGES_TARGET) rev-list --max-parents=0 gh-pages`; \
	if [ `git -C $(GHPAGES_ROOT) show -s --format=%ct $$ROOT` -lt $$CUTOFF ]; then \
	  git -C $(GHPAGES_ROOT) replace --graft `git rev-list --min-age=$$KEEP --max-count=1 gh-pages` && \
	  git -C $(GHPAGES_ROOT) filter-branch gh-pages; \
```

The renaming and switch to backticks is merely an attempt to get this consistent.  The && is important.  The cleanup happens before we add new commits and if the repo has been idle for the last 30 days, I don't want to generate an error.

---

I _think_ this should already work on a 30-day-idle repo, since it chooses the first commit older than 30 days as the one to keep.  If there's been no activity for 30 days, then however old the most recent commit is, that's the commit that will be chosen as the new beginning-of-time for the gh-pages branch, and the new state will be added as the child.